### PR TITLE
ROX-22551: Cleanup Fleet Manager after the addon CI/CD implementation

### PR DESCRIPTION
**Pull request by kovayur**
## Description
Removing dead code I found while implementing the addon integration

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Added test description under `Test manual`
- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [ ] CI and all relevant tests are passing
- [ ] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.
- [ ] Add secret to app-interface Vault or Secrets Manager if necessary
- [ ] RDS changes were e2e tested [manually](../docs/development/howto-e2e-test-rds.md)
- [ ] Check AWS limits are reasonable for changes provisioning new resources
- [ ] (If applicable) Changes to the dp-terraform Helm values have been reflected in the addon on integration environment

## Test manual

**TODO:** Add manual testing efforts

```
# To run tests locally run:
make db/teardown db/setup db/migrate
make ocm/setup
make verify lint binary test test/integration
```

### CODE REVIEW

File: pkg/services/sso/iam_service.go [hunk 1]
```diff
@@ -1,37 +0,0 @@
...
-	ClientID       string
-	Name           string
-	Description    string
-}
-
-// IAMService ...
-//
-//go:generate moq -out iam_service_moq.go . IAMService
-type IAMService interface {
-	RegisterAcsFleetshardOperatorServiceAccount(agentClusterID string) (*api.ServiceAccount, *errors.ServiceError)
```
Comment (kovayur) [inline pkg/services/sso/iam_service.go:28]: Remove the dynamic creation of the fleet manager service account

**Review by ebensh — APPROVED** [review]

**Comment by johannes94** [comment]
trigger Jira automation

**Pull request merged**